libfuse/dir.go

```diff
@@ -134,6 +134,26 @@ func (f *Folder) forgetNode(node libkbfs.Node) {
 
 var _ libkbfs.Observer = (*Folder)(nil)
 
+func (f *Folder) resolve(ctx context.Context) (*libkbfs.TlfHandle, error) {
```

Comment:
You could factor this out into `libfs` and have it take a `libkbfs.Observer` in order to call `TlfHandleChange`.